### ISSUE TITLE
Rumble never stops on Gulikit King Kong 2 Pro Bluetooth controller

### ISSUE THREAD

**Issue by lionsnob** [ChimeraOS/chimeraos#420]
Hello -

After I pair this controller in X-input mode rumble turns on and never shuts off. Is there anything I can do to stop this behavior? Any log I can provide?

**Comment by Samsagax** [comment]
Does this happens on latest build?

**Comment by karbowiak** [comment]
Just installed latest version from https://chimeraos.org/download and it also happens to me with a GuliKit controller.

I have tried the "D" mode as well, but it never connects.

I tried joining Discord using the link here https://github.com/ChimeraOS/chimeraos/wiki/FAQ
But it seems to be dead atm :(

On initial boot i could get root and see dmesg, but after a reboot that went away as well (tried using the password i set via the webui, and gamer, and a few others)

Funny thing is, the GuliKit controller worked perfectly on HoloISO 😅

**Comment by pastaq** [comment]
hid-nintendo driver was added to the kernel in 5.16 and updated in 6.1. Since HoloISO is using 5.13 they may be using another driver. 

We only have one Nintendo related package, [nintendo-udev](https://aur.archlinux.org/cgit/aur.git/tree/70-nintendo.rules?h=nintendo-udev), which may or may not still be required. If you temporarily move `/usr/lib/udev/rules.d/70-nintendo.rules` does the issue go away? I can test with my official pro controller later to see what that does.

**Comment by karbowiak** [comment]
@pastaq 

I dropped a load of text in Discord in #general

TLDR: It seems the GuliKit loads the xpadneo driver, and it has a bug - i tried adding the following to the xbox-pads.rules quirk located here: https://github.com/ChimeraOS/device-quirks/blob/main/usr/lib/udev/rules.d/60-xbox-pads.rules

```
# Gulikit KingKong 2 Pro (Bluetooth)
ATTRS{idVendor}=="045e", ATTRS{idProduct}=="028e", RUN+="/sbin/modprobe xpad" RUN+="/bin/sh -c 'echo 045e 028e > /sys/bus/usb/drivers/xpad/new_id'"
ATTRS{idVendor}=="28de", ATTRS{idProduct}=="11ff", RUN+="/sbin/modprobe xpad" RUN+="/bin/sh -c 'echo 28de 11ff > /sys/bus/usb/drivers/xpad/new_id'"
```

And after a few reboots, it seemed to kick in, and load the xpad driver for the GuliKit KingKong 2 Pro.

@lionsnob can you confirm?

**Comment by sabatmonk** [comment]
Still rumbling constantly using this fix (unless it requires more than 5 reboot and "udevadm control --reload-rules && udevadm trigger")

**Comment by pastaq** [comment]
> Still rumbling constantly using this fix (unless it requires more than 5 reboot and "udevadm control --reload-rules && udevadm trigger")

What is the output when you run `cat /proc/bus/input/devices`

**Comment by lionsnob** [comment]
I wonder if this is fixed in kernel 6.4?

https://www.phoronix.com/news/Linux-6.4-HID

**Comment by pastaq** [comment]
Does this device have multiple input modes like other gulikit devices? D-input vs X-input, etc?

**Comment by karbowiak** [comment]
@pastaq it does, you can switch between..
- Direct
- Android / iOS
- X
- Nintendo

XW is as far as i remember, the X input mode (I am not currently on linux or chimeraOS, so operating from memory here 😅)

**Comment by byronpc** [comment]
is there any update regarding this?

**Comment by ChaosBlades** [comment]
Well this bug is annoying. This controller works perfectly on HoloISO and Pop_OS. ChimeraOS also didn't see my Xbox One S controller with Bluetooth until I tried to pair it in Gnome.

**Comment by FabricioMSantos** [comment]
The problem still happens on 44 stable

**Comment by jackgraddon** [comment]
Still happening currently. When switching to DX mode, button maps are completely screwed up. iOS/Android mode does the same as XW. Nintendo mode does work and is mapped correctly, but it loses analog triggers (because a real pro controller doesn’t have analog triggers).

**Comment by ChaosBlades** [comment]
This has been fixed with the release of ChimeraOS 45 (013ff33) Pre-release from yesterday. In the testing branch.
Feel free to close the issue.

**Comment by pastaq** [comment]
We'll close it when we release v45. Thanks for testing.
